File: solutions/csharp/lucians-luscious-lasagna/2/LuciansLusciousLasagna.cs
class Lasagna
{
    public int ExpectedMinutesInOven() => 40;

    public int RemainingMinutesInOven(int elapsed) 
    {
        return ExpectedMinutesInOven() - elapsed;
    }

    public int PreparationTimeInMinutes(int layers) => 2 * layers;

    public int ElapsedTimeInMinutes(int layers, int timeInOven)
    {
        return timeInOven + PreparationTimeInMinutes(layers);
    }
}

File: solutions/fsharp/lucians-luscious-lasagna/1/LuciansLusciousLasagna.fs
module LuciansLusciousLasagna

let expectedMinutesInOven = 40

let remainingMinutesInOven elapsed = expectedMinutesInOven - elapsed

let prepTimePerLayer = 2

let preparationTimeInMinutes layers = prepTimePerLayer * layers

let elapsedTimeInMinutes layers elapsed = 
    elapsed + preparationTimeInMinutes layers
